### DISCUSSION TITLE
Accelerate was not able to connect to your database. The underlying error is: Health Check of Query Engine timed out.

### DISCUSSION THREAD

**Discussion by akintolafemi** [prisma/prisma#22375] [Q&A]
### Question

Hi everyone. Please I am new to Prisma Accelerate and experimenting with a new project. I have followed the documentation on installation and configuration step by step and I am sure I have done everything right. However, when I run the login endpoint that queries users table, I get the following error message 

This request could not be understood by the server: {"type":"UnknownJsonError","body":{"code":"P6008","message":"Accelerate was not able to connect to your database. The underlying error is: Health Check of Query Engine timed out."}}

These are the Prisma versions I am on
```
"@prisma/client": "^5.7.0",
"@prisma/extension-accelerate": "^0.6.2",
```
Database is MariaDB 11 with MySQL.

In my .env
```
DATABASE_URL=prisma://accelerate.prisma-data.net/?API_KEY
DIRECT_DATABASE_URL=mysql://user:password@localhost:3306/mydb
```
In my schema.prisma file
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
  directUrl = env("DIRECT_DATABASE_URL")
}

model users {
  id        Int        @id @default(autoincrement()) @db.UnsignedInt
  username  String     @unique @db.VarChar(50)
  password  String     @db.VarChar(100)
  createdat DateTime   @default(now())
}
```
And lastly, my prismaclient class
```ts
import {
  Injectable,
  OnModuleInit,
  INestApplication,
  Logger,
} from '@nestjs/common';
import { PrismaClient } from '@prisma/client';
import { withAccelerate } from '@prisma/extension-accelerate'

@Injectable()
export class PrismaService extends PrismaClient implements OnModuleInit {
  constructor() {
    super({
      // log: ['query']
    });
  }
  async onModuleInit() {
    await this.$extends(withAccelerate()).$connect();
    new Logger(PrismaService.name).log('Connected to MySQL service database');
  }

  async enableShutdownHooks(app: INestApplication) {
    process.on("beforeExit", async () => {
      await app.close();
    });
  }
}
```
My database is remotely accessible

Thank you

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by nurul3101** [comment]
Hey @akintolafemi 👋 

Thank you for sharing your question.

To ensure I understand your setup correctly, your database is publicly accessible, right?

Do you get this error intermittently, or are all queries failing with this error?

**Reply by akintolafemi** [reply]
@nurul3101 
thanks for replying. Yes, the database is accessible from mysql workbench and if I use the db connection string mysql://user:password@localhost:3306/mydb instead of prisma://accelerate.prisma-data.net/?API_KEY. 

And the error is constant. All queries fail with this same error

Also, on https://console.prisma.io for the project workspace, I can see that the insight changes each time I send a query.

**Reply by nurul3101** [reply]
@akintolafemi Thanks for the follow-up reply. Referring to your connection string - "mysql://user:password@**localhost**:3306/mydb ", is your database running locally? Or was it just an example?
If you are using a db provider, can you share which one?

(cc @ludralph)

**Reply by akintolafemi** [reply]
Hi @nurul3101 
The connection string I provided was just an example. I am running MariaDB (version 11) running on a Namecheap VPS. And it is publicly accessible

**Comment by alienbuild** [comment]
I have the same issue and suspect its actually my server blocking prisma accelerate after speaking with discord staff. I'm just trying to find out what I need to allow in my server for prisma accelerate to connect. Assuming the requests to connect to my database are coming from `accelerate.prisma-data.net` I've allowed it in my additional mysql access hosts (Whm), but it hasn't worked.

**Reply by nurul3101** [reply]
@alienbuild Are you using MariaDB as well? Which DB hosting provider are you using? We would like to reproduce it to investigate this. Is your database publicly accessible?

**Reply by alienbuild** [reply]
Thank you for the reply. Yes its MariaDB 10.5.23 and publicly accessible hosted via liquidweb.com

**Reply by nurul3101** [reply]
Thanks for confirming. I am wondering if this issue is MariaDB specific.

**Reply by akintolafemi** [reply]
@nurul3101 I think it could be MariaDB specific. From prisma documentation, in the list of supported databases, it states MariaDB is supported but up to version 10 only. [https://www.prisma.io/docs/orm/reference/supported-databases](url). And in my case, I'm using MariaDB 11.2.2. Been wondering if I should downgrade to 10 but I need to be sure that;s the problem before downgrading

**Reply by alienbuild** [reply]
I'm using version 10 and have the same issue so that probably won't resolve much for you to be honest. 

@nurul3101 I've created a mini test application for you here 
https://github.com/alienbuild/prisma-test

and I've sent you the .env with the database credentials on slack private messages.

Additionally I've also tried adding the wildcard '%' into the sql remote hosts config, but no results.

**Reply by alienbuild** [reply]
Update from Ankur today is the following


> Hey there , 
> 
> Just wanted to share a quick update. We've successfully identified the underlying bug causing the issue you reported. We've outlined the necessary steps to address it, although I can't provide an exact timeframe for resolution at the moment.
> 
> As soon as we make progress and complete the necessary steps, I'll reach out to keep you informed. Thanks for your understanding! 🚀😊

**Comment by akintolafemi** [comment]
@alienbuild @nurul3101 anyone got a solution to this or what the issue is?

**Reply by alienbuild** [reply]
Still waiting for a solution. I've got a discord ticket open with @nurul3101 and @ankur-arch who are looking into it, but I think the Christmas and new years period has probably delayed any progress at the moment.

**Reply by akintolafemi** [reply]
thanks @alienbuild

**Reply by nurul3101** [reply]
@alienbuild Thank you for your patience. We are now actively looking into this. Is the database still accessible?
We aren't able to connect to it.
(We can continue the conversation on Discord if its easier)

**Reply by akintolafemi** [reply]
@nurul3101 @alienbuild kindly share the discord link so I can join the conversation. Thanks

**Reply by ankur-arch** [reply]
here's the [link](https://discord.com/channels/937751382725886062/1149992975099772988) to the conversation. cc: @akintolafemi

**Reply by akintolafemi** [reply]
Hi @ankur-arch thank you. But I'm unable to join the conversation. 
<img width="1344" alt="Screenshot 2024-01-13 at 21 02 37" src="https://github.com/prisma/prisma/assets/22343293/01760775-a74f-4ec3-8e4b-225cbc172ede">

**Reply by ankur-arch** [reply]
@akintolafemi can you open a new question, and tag me, mentioning the details of the issue you're facing; this issue is still under investigation. Once, we come to a conclusion, I can communicate the findings to you.

**Reply by akintolafemi** [reply]
@ankur-arch Good day. I actually opened this current question here. So I just wanted to follow the discord conversation so I can pick one or two things.

**Reply by akintolafemi** [reply]
Hi guys @nurul3101 @alienbuild @ankur-arch 
Any updates on this issue?

**Reply by nurul3101** [reply]
Hey @akintolafemi 👋 

There is unfortunately no update at the moment. Our team has confirmed the issue. I'll inform you as soon as there is an update from our end. 

Thank you for your patience while we work on resolving this 🙏

**Reply by TheGreyRaven** [reply]
> Hey @akintolafemi 👋
> 
> There is unfortunately no update at the moment. Our team has confirmed the issue. I'll inform you as soon as there is an update from our end.
> 
> Thank you for your patience while we work on resolving this 🙏

Thats unfortunate, might be a bit of a long shot but is there an experimental branch or something similar that fixes or attempts to fix this issue?
I currently have a large project that relies on MariaDB and right now this is issue is blocking our production build and we have to restort to another ORM as solution.

Thanks for the hard work you guys put in!

**Comment by fwx5618177** [comment]
In my opinion, I suspect the data-proxy cannot get the blank result.

**Comment by mundume** [comment]
this just happened to me right now and everything was working a second ago

**Reply by jacklynch00** [reply]
same

**Comment by akintolafemi** [comment]
Hi @nurul3101 

I thought to come back to this again today and still encountered same problem. I don't know if there's been any updates on this as of now?

cc: @ankur-arch @alienbuild

**Comment by davi0812** [comment]
Hello, going thru the same, when using the database url without accelerate works fine, as soon as i change that to direct url and use the prisma accelerate one for database it can't connect.

**Reply by ludralph** [reply]
Hi @davi0812 

Can you please open a new discusssion. Thank you 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
